### PR TITLE
Continued tests and features.

### PR DESCRIPTION
**Pull request by achedeuzot**

### CODE REVIEW

File: tartiflette/executors/basic.py [hunk 1]
```diff
@@ -2,15 +2,15 @@
 
 
 async def _level_execute(resolvers, request_ctx):
-    corroutines = [
+    coroutines = [
```
Comment (abusi) [inline tartiflette/executors/basic.py]: :+1:

File: tartiflette/parser/nodes/field.py [hunk 2]
```diff
@@ -46,7 +47,7 @@ def _to_dict(thing):
 
 class NodeField(Node):
     def __init__(self, func, location, path, name, type_condition):
-        super(NodeField, self).__init__(path, 'Field', location, name)
+        super().__init__(path, 'Field', location, name)
```
Comment (abusi) [inline tartiflette/parser/nodes/field.py]: IMHO it's more consistent to keep the super fully explicit, cause if we double inherit sometimes, it won't creates confusion. Explicit > *.

File: tartiflette/parser/visitor.py [hunk 3]
```diff
@@ -66,16 +68,17 @@ def __init__(self, variables=None, schema_definition=None):
...
 
-    def _on_argument_in(self, element):
+    def _on_argument_in(self, element: _VisitorElement):
         self._current_argument_name = element.name
 
     def _on_argument_out(self, _):
         self._current_argument_name = None
 
-    def _on_value_in(self, element):
+    def _on_value_in(self, element: _VisitorElement):
```
Comment (abusi) [inline tartiflette/parser/visitor.py]: Nice, I want to cythonify all this at some point :).

File: tartiflette/parser/visitor.py [hunk 4]
```diff
@@ -107,32 +110,31 @@ def _on_variable_in(self, element):
...
-            # KeyError no resolver
-            pass
+        field_name = parent_name + '.' + element_name
+        field = self._schema_definition.get_field_by_name(field_name)
 
-        return _default_resolver
+        if field:
+            self._parent_node_type = reduce_type(field.gql_type)
+            return field.resolver
+        else:
```
Comment (abusi) [inline tartiflette/parser/visitor.py]: can you try: except: this instead of lybl ?

File: tartiflette/resolver_decorator.py [hunk 5]
```diff
@@ -24,6 +38,13 @@ def __init__(self, name: str, schema=None):
         self.field = self._schema.get_field_by_name(name=name)
 
     def __call__(self, resolver, *args, **kwargs):
+        if not iscoroutinefunction(resolver):
```
Comment (abusi) [inline tartiflette/resolver_decorator.py]: I thing it's moot to test IF we decide that in order to work in our engine, a resolver MUST/SHOULD be async ? @tsunammis what do you thing ?

File: tartiflette/resolver_decorator.py [hunk 6]
```diff
@@ -24,6 +38,13 @@ def __init__(self, name: str, schema=None):
...
+            raise TartifletteNonAwaitableResolver(
+                "The resolver `{}` given for the field `{}` "
+                "is not awaitable.".format(repr(resolver), self.field.name))
         if self.field:
+            # self.field.resolver = _
+            # ResolverWrapper(self._schema.get_type(
+            # self.field.gql_type).to_primitive, resolver)
```
Comment (abusi) [inline tartiflette/resolver_decorator.py]: this 
```
if field:
   self.field.resolver = resolver
```   should be 
```
try:
    self.field.resolver = resolver
except AttributeError:
    pass
```

File: tartiflette/schema.py [hunk 7]
```diff
@@ -104,8 +104,18 @@ def subscription_type(self) -> Optional[str]:
...
+        return self._gql_types
+
     def add_definition(self, value: GraphQLType) -> None:
         # TODO: Check stuff, call update_schema after each new definition ?
+        if not isinstance(value, GraphQLType):
+            raise TartifletteSchemaValidationError(
+                'new GraphQLSchema type definition `{}` '
+                'is not a GraphQLType.'.format(
+                    value.__class__.__name__)
+            )
```
Comment (abusi) [inline tartiflette/schema.py]: maybe hanlde the method in a try-except and raise the TfftValidError if AttributeError is raised ?

File: tartiflette/types/enum.py [hunk 8]
```diff
@@ -17,16 +17,17 @@ class GraphQLEnumValue:
     def __init__(self, value: Any = None, name: Optional[str]=None, description: Optional[str]=None):
         self.value = value
         self.name = str(name)
+        if not name:
```
Comment (abusi) [inline tartiflette/types/enum.py]: also don't do str(name) if name is none :)

File: tartiflette/parser/nodes/field.py (hunk 2 above)
Comment (achedeuzot) [reply tartiflette/parser/nodes/field.py]: As seen with bubu, it's OK. https://stackoverflow.com/questions/26927571/multiple-inheritance-in-python3-with-different-signatures :D

File: tartiflette/parser/visitor.py (hunk 3 above)
Comment (achedeuzot) [reply tartiflette/parser/visitor.py]: That would be amazing :D

File: tartiflette/resolver_decorator.py (hunk 5 above)
Comment (achedeuzot) [reply tartiflette/resolver_decorator.py]: This is to filter the inputs of the user. We won't check afterwards :)

File: tartiflette/resolver_decorator.py (hunk 6 above)
Comment (achedeuzot) [reply tartiflette/resolver_decorator.py]: 👍

File: tartiflette/schema.py (hunk 7 above)
Comment (achedeuzot) [reply tartiflette/schema.py]: Removed this safety. If someone messes with the schema, it's his responsibility.

File: tartiflette/types/enum.py (hunk 8 above)
Comment (achedeuzot) [reply tartiflette/types/enum.py]: Enum has to be updated / reworked anyway :x

File: tartiflette/parser/visitor.py (hunk 4 above)
Comment (achedeuzot) [reply tartiflette/parser/visitor.py]: Fixed in last commit :)

**Pull request merged**